client/src/components/common/boost/BoostConfigurator.tsx

```diff
@@ -23,17 +25,49 @@ export default function BoostConfigurator({
   onDeleteConfig,
   onUploadConfig,
 }: BoostConfiguratorProps) {
+  const fileInput = createRef<HTMLInputElement>();
+  const [configType, setConfigType] = useState<BoostConfigType>('all');
+
+  // Function to click the hidden file input button (this is a work around to avoid the ugly
+  // UI of the default input file button)
+  const handleClick = (type: BoostConfigType) => {
+    if (fileInput.current) {
+      setConfigType(type);
+      fileInput.current.click();
+    }
+  };
+
+  const handleFileUpload = (event: React.ChangeEvent<HTMLInputElement>) => {
+    // This method gets called even if the user uploads a file and the second time they cancel
+    // to upload, hence the need to check that the files attribute is not an array of 0 length
+    if (event.target.files != null && event.target.files.length !== 0) {
+      onUploadConfig(configType, event.target.files);
+    }
+  };
+
   return (
-    <Card>
+    <Card style={{ marginTop: '2.5rem' }}>
       <Card.Body>
         <Card.Title>Configuration</Card.Title>
-        <Button
-          variant="outline-primary"
-          className="mb-3"
-          onClick={onUploadConfig}
-        >
-          Upload config
-        </Button>
+        <>
+          <input
+            ref={fileInput}
+            onChange={handleFileUpload}
+            type="file"
+            style={{ display: 'none' }}
+            accept=".json"
+            multiple={false}
+          />
+          <Button
+            variant="outline-primary"
+            className="mb-3"
+            onClick={() => {
+              handleClick('all');
+            }}
+          >
+            Upload All Configs
```

Comment:
I reckon this button could look neater if it were on the right side of the card, in line with the other upload buttons (and title, saving space). If putting it on the same row as the card title is too difficult don't worry though, probably best where it is otherwise we'll have an awkward amount of empty space.

Reply:
<img width="739" alt="Screen Shot 2021-02-25 at 1 40 34 pm" src="https://user-images.githubusercontent.com/63642262/109095054-21f96600-776f-11eb-8802-4b085eecb5fc.png">
Does this look better? I tried aligning 'Configuration' to be a little bit vertically entered but all my tries didn't work lol.

Reply:
Nah this looks great!